File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int counter = 0;
        int j = 0;

        for (int num: nums) {

            if (num == 0) {
                counter++;
            }

            if(num != 0) {
                nums[j] = num;
                j++;
            }

        }

        for(int k = 0; k < counter; k++) {
            nums[j] = 0;
            j++;
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.swift
class Solution {
    func moveZeroes(_ nums: inout [Int]) { //the inout allows for the inputs to be modified -- without, its passed in as a copy

      //write pointer - where the next non-zero value should be placed
      var write = 0;

      //iterate thru array
      for read in 0..<nums.count {
        
        //if current is a nonzero, then have to swap it
        if nums[read] != 0 {
          nums.swapAt(write, read)
          write = write + 1
        }
     }
  }
}